bigflow/testing/isolate.py

```diff
@@ -0,0 +1,218 @@
+"""Mixins to isolate `unittest.TestCase` into separate processes
+
+May prevent from doing side effects with Python intepreter
+(loaded modules, global variables, caches etc).
+"""
+
+import unittest
+import sys
+import time
+import contextlib
+import multiprocessing
+import threading
+
+
+class _ObjectTag:
+    """Stub for non-pickable objects, which needed to be transferred between processes"""
+
+    def __init__(self, tag):
+        self.tag = tag
+
+    @classmethod
+    def forfeit(cls, obj, obj2tag):
+        return cls(obj2tag[obj]) if obj in obj2tag else obj
+
+    @classmethod
+    def recover(cls, obj, tag2obj):
+        return tag2obj[obj.tag] if isinstance(obj, cls) else obj
+
+
+class _TestResultProxy:
+    """Dump proxy for `TestResult`. Stream all public method calls into channel."""
+
+    def __init__(self, ch, testcase_proxy, tblib):
+        self._ch = ch
+        self._tc_proxy = testcase_proxy
+        self._tblib = tblib
+        self._lock = threading.Lock()
+
+    # proxy all public method calls
+    def __getattr__(self, name: str):
+        assert not name.startswith("_")
+
+        def method(*args):
+            with self._lock:
+                # replace ref to proxy testcase by sepulka
+                smap = {self._tc_proxy: 'testcase'}
```

Comment:
smap?

Reply:
**S**ubstitute**MAP**
This is only a local variable, used only the next line (so effective variable scope is 2 nearest lines).